src/Raven.Client/Documents/Session/Operations/BatchOperation.cs

```diff
@@ -102,6 +112,24 @@ public void SetResult(BatchCommandResult result)
                 var afterSaveChangesEventArgs = new AfterSaveChangesEventArgs(_session, documentInfo.Id, documentInfo.Entity);
                 _session.OnAfterSaveChangesInvoke(afterSaveChangesEventArgs);
             }
+
+        }
+
+        private void UpdateCounterValuesInCache(BlittableJsonReaderArray counters, string docId)
+        {
+            if (_session.CountersByDocId.TryGetValue(docId, out InMemoryDocumentSessionOperations.CountersCache cache) == false)
+            {
+                cache = new InMemoryDocumentSessionOperations.CountersCache();
```

Comment:
You didn't _add_ it back to the `CountersByDocId` ?

Reply:
fixed